src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidator.cs

```diff
@@ -71,5 +83,31 @@ public IEnumerable<ModelValidationResult> Validate(ModelValidationContext valida
 
             return Enumerable.Empty<ModelValidationResult>();
         }
+
+        private string GetErrorMessage(ModelValidationContextBase validationContext)
+        {
+            if (validationContext == null)
+            {
+                throw new ArgumentNullException(nameof(validationContext));
+            }
+            if (validationContext.ModelMetadata == null)
+            {
+                throw new ArgumentNullException(nameof(validationContext.ModelMetadata));
+            }
+            if (validationContext.MetadataProvider == null)
+            {
+                throw new ArgumentNullException(nameof(validationContext.MetadataProvider));
```

Comment:
Don't `throw` an `ArgumentNullException` unless the argument itself is `null`. Use an `ArgumentException` with `Resources.FormatPropertyOfTypeCannotBeNull()`. Might need to copy that resource from another project in this repo.